src/lib.rs

```diff
@@ -270,14 +305,24 @@ impl SyncEditorSystem {
     }
 }
 
-impl Default for SyncEditorSystem {
-    fn default() -> Self { SyncEditorSystem::new() }
-}
-
 impl<'a> System<'a> for SyncEditorSystem {
     type SystemData = Entities<'a>;
 
     fn run(&mut self, entities: Self::SystemData) {
+        // Determine if we should send full state data this frame.
+        let now = Instant::now();
+        let send_this_frame = now > self.next_send;
+
+        // Calculate when we should next send full state data.
+        //
+        // NOTE: We iteratively advance the next send time by `send_interval` in order to ensure
+        // we send at a regular cadence. We also ensure that the next send time is after `now`
+        // to avoid running into a death spiral if a frame spike causes frame time to be so long
+        // that the next send time would still be in the past.
+        while self.next_send < now {
+            self.next_send += self.send_interval;
```

Comment:
According to your note i would expect now+send_interval

Reply:
`next_send += send_interval` is what I intended. Doing `now + send_interval` would cause the cadence to gradually drift since the timing for when the frames run isn't exact. I'll update the wording of the comments to try to make this more clear :)